src/rdkafka_cgrp.c

```diff
@@ -1573,27 +2010,34 @@ void rd_kafka_cgrp_handle_Heartbeat (rd_kafka_t *rk,
                 if (rkcg->rkcg_join_state ==
                     RD_KAFKA_CGRP_JOIN_STATE_WAIT_REVOKE_REBALANCE_CB)
                         return;
-                rebalance_reason = "group is rebalancing";
-                break;
+                rd_kafka_cgrp_group_is_rebalancing(rkcg);
+                return;
 
         case RD_KAFKA_RESP_ERR_UNKNOWN_MEMBER_ID:
                 rd_kafka_cgrp_set_member_id(rkcg, "");
-                rebalance_reason = "resetting member-id";
-                assignment_lost = rd_true;
-                break;
+                rd_kafka_cgrp_revoke_all_rebalance(rkcg,
+                                                   rd_true/*lost*/,
+                                                   rd_true/*initiating*/,
+                                                   "resetting member-id");
+                return;
 
         case RD_KAFKA_RESP_ERR_ILLEGAL_GENERATION:
-                rebalance_reason = "group is rebalancing";
-                assignment_lost = rd_true;
-                break;
+                rd_kafka_cgrp_revoke_all_rebalance(rkcg,
+                                                   rd_true/*lost*/,
+                                                   rd_true/*initiating*/,
+                                                   "illegal generation");
```

Comment:
this wording is taken care of by debug messages in `rd_kafka_cgrp_rebalance_revoke_all`. additionally, I differentiate "group is rebalancing" from "initiating rebalance", since the former wording, although it could be argued is correct, may imply that it is not this member that caused the rebalance, which may be confusing. Regardless, differentiating the cases adds information content to the logging.

I spent some time confused by the wording here when reading logs, and the changes are addressing that confusion.